libminifi/src/utils/OsUtils.cpp

```diff
@@ -154,6 +160,80 @@ std::string OsUtils::userIdToUsername(const std::string &uid) {
   return name;
 }
 
+unsigned long long OsUtils::getMemoryUsage() {
+#ifdef __linux__
+  long resPages;
+  long sharedPages;
+  {
+    std::string ignore;
+    std::ifstream ifs("/proc/self/statm");
+    ifs >> ignore >> resPages >> sharedPages;
+  }
+
+  if (sharedPages > resPages) {
+    throw std::range_error("Shared memory page count ("
+      + std::to_string(sharedPages)
+      + ") should not be larger than resident set size ("
+      + std::to_string(resPages)
+      + "), that includes it"
+    );
+  }
+
+  const long ownPages = resPages - sharedPages;
```

Comment:
Please do not use long!
Long is 32 bits on Win and 64 on Linux.

Moreover the multiplication below has a high chance of overflowing.

Reply:
This code is Linux-only. Man says values in /proc/self/statm will be long: https://man7.org/linux/man-pages/man5/proc.5.html. Page size is usually 4096, memory usage value should definitely fit 64 bytes even after multiplication.

Reply:
The linter will also report for `long` usage.